security/security-core/src/main/java/io/camunda/security/auth/OidcGroupsLoader.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.security.auth;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.InvalidPathException;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import com.jayway.jsonpath.spi.cache.CacheProvider;
+import com.jayway.jsonpath.spi.cache.LRUCache;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class OidcGroupsLoader {
+  public static final String CLAIM_NOT_STRING_OR_STRING_ARRAY =
+      "Configured claim for %s (%s) is not a string or string array. Please check your OIDC configuration.";
+
+  private static final Configuration CONFIGURATION =
+      Configuration.builder()
+          // Ignore the common case that the last path element is not set
+          .options(Option.DEFAULT_PATH_LEAF_TO_NULL)
+          .jsonProvider(null)
+          .mappingProvider(null)
+          .build();
+
+  static {
+    CacheProvider.setCache(new LRUCache(100));
```

Comment:
❓ I noticed that in the other loaders we don't use a cache (I think Lena showed it was pretty performant) is there a real concern/need here for the cache at the moment?

Reply:
Adding to this:
1. Using a cache at all is not safe because compiled expressions are not thread safe
2. This is modifying global state which is pretty bad. It's not testable and it impacts _every_ use of JsonPath

Reply:
I've added the cache becuase I had some local test to check the thread-safety and wanted to see how it affected, then It leaked into my commit.